src/libs/ttf/lv_ttf.c

```diff
@@ -0,0 +1,4512 @@
+#define STB_RECT_PACK_IMPLEMENTATION
+#define STBRP_STATIC
+#include <lv_ttf.h>
+#include <stdio.h>
+
+#include "stb_rect_pack.h"
+#define STBTT_STATIC
+#include "stb_truetype_htcw.h"
+
+// e.g. #define your own STBTT_ifloor/STBTT_iceil() to avoid math.h
+#ifndef STBTT_ifloor
+#include <math.h>
+#define STBTT_ifloor(x) ((int)floor(x))
+#define STBTT_iceil(x) ((int)ceil(x))
+#endif
+
+#ifndef STBTT_sqrt
+#include <math.h>
+#define STBTT_sqrt(x) sqrt(x)
+#define STBTT_pow(x, y) pow(x, y)
+#endif
+
+#ifndef STBTT_fmod
+#include <math.h>
+#define STBTT_fmod(x, y) fmod(x, y)
+#endif
+
+#ifndef STBTT_cos
+#include <math.h>
+#define STBTT_cos(x) cos(x)
+#define STBTT_acos(x) acos(x)
+#endif
+
+#ifndef STBTT_fabs
+#include <math.h>
+#define STBTT_fabs(x) fabs(x)
+#endif
+
+// #define your own functions "STBTT_malloc" / "STBTT_free" to avoid malloc.h
+#ifndef STBTT_malloc
+#include <stdlib.h>
+#define STBTT_malloc(x, u) ((void)(u), lv_mem_alloc(x))
+#define STBTT_free(x, u) ((void)(u), lv_mem_free(x))
+#endif
+
+#ifndef STBTT_assert
+#include <assert.h>
+#define STBTT_assert(x) assert(x)
+#endif
+
+#ifndef STBTT_strlen
+#include <string.h>
+#define STBTT_strlen(x) strlen(x)
+#endif
+
+#ifndef STBTT_memcpy
+#include <string.h>
+#define STBTT_memcpy memcpy
+#define STBTT_memset memset
+#endif
+
+#ifndef STBTT_MAX_OVERSAMPLE
+#define STBTT_MAX_OVERSAMPLE 8
+#endif
+
+#if STBTT_MAX_OVERSAMPLE > 255
+#error "STBTT_MAX_OVERSAMPLE cannot be > 255"
+#endif
+
+typedef int stbtt__test_oversample_pow2[(STBTT_MAX_OVERSAMPLE & (STBTT_MAX_OVERSAMPLE - 1)) == 0 ? 1 : -1];
+
+#ifndef STBTT_RASTERIZER_VERSION
+#define STBTT_RASTERIZER_VERSION 2
+#endif
+
+#ifdef _MSC_VER
+#define STBTT__NOTUSED(v) (void)(v)
+#else
+#define STBTT__NOTUSED(v) (void)sizeof(v)
+#endif
+
+static stbtt_uint8 read_uint8(lv_fs_file_t *stream) {
+    uint8_t result;
+    uint32_t br;
+    lv_fs_read(stream, &result, 1, &br);
+    return result;
+}
+static stbtt_int8 read_int8(lv_fs_file_t *stream) {
+    int8_t result;
+    uint32_t br;
+    lv_fs_read(stream, (uint8_t *)&result, 1, &br);
+    return result;
+}
+// TODO: Support big endian procs
```

Comment:
Not critical, just FYI LVGL has the `LV_BIG_ENDIAN_SYSTEM` config.

Reply:
I meant to ask about that (hence the TODO). I had one in htcw_gfx as well.